### PR TITLE
feat: orchestrate workflows

### PR DESCRIPTION
**Pull request by RaitoBezarius**
This avoids double building needlessly.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 name: Deployments
 on:
+  # Run the deployment only once staging has been prepared.
+  workflow_run:
+    workflows: ["Trusted checks"]
```
Comment (fricklerhandwerk) [inline .github/workflows/deploy.yaml:5]: I got confused by those names, had to read everything again to make sense of it. Can we call them "build" and "dry-activate" so it's clear what they're doing? One check being more "trusted" than the other doesn't convey much meaning.

**Review by fricklerhandwerk — APPROVED** [review]
Logic looks good, only needs fixing [`Left Karlton`](https://martinfowler.com/bliki/TwoHardThings.html)

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (RaitoBezarius) [reply .github/workflows/deploy.yaml]: @fricklerhandwerk This is done.

**Review by fricklerhandwerk — APPROVED** [review]
Yes, nice, thanks for the extra explanatory comments

**Pull request merged**